### DISCUSSION TITLE
REST Status Code List Query

### DISCUSSION THREAD

**Discussion by macntech** [directus/directus#15056] [Archive]
Hello,
not sure if that has been discussed yet. I am just checking the list routes Directs provides. E.g. if I search for a user based on the email address I would query the API like

https://URL/users?filter[email][_eq]=email@email.com

I am just wondering if the status code return is correct. Directus returns code 200 + user information if the user has been found. BUT directus also returns 200 if no user has been found just with an empty body. Should it not be more sufficient if the API would return e.g. 204 or even 404 so we can catch the error more straight forward in frontend apps? 

<img width="929" alt="Bildschirmfoto 2022-08-13 um 2 08 10 PM" src="https://user-images.githubusercontent.com/50032663/184493391-21626894-a92c-41ef-b2c1-07c29627721e.png">

**Comment by azrikahar** [comment]
> Should it not be more sufficient if the API would return e.g. 204 or even 404 so we can catch the error more straight forward in frontend apps?

The API does return 204 when there's no content, thanks to #10569! However in this scenario, it's not exactly no content, but an empty `"data"` array, so it's technically working as intended.

I'm not aware of actual specs for this kind of scenarios, but personally I do agree it'd make more sense for "no user has been found" to be applicable when we're doing `/users/<uuid>` for example, but not listing endpoints such as `/users` (even with filters). But of course that's up for discussion!

As for clearer error codes such as 404 which you may notice will result as 403 at times, this is being discussed in #4368 👍

**Comment by rijkvanzanten** [comment]
IMO 200 is the correct response here. The API _did_ respond with a body, and in this case, the successful result of your search operation is an empty array of users 🤔
